packages/next/client/image.tsx

```diff
@@ -291,6 +320,17 @@ export default function Image({
   const heightInt = getInt(height)
   const qualityInt = getInt(quality)
 
+  const MIN_IMG_SIZE_FOR_PLACEHOLDER = 100
+  const tooSmallForBlurryPlaceholder =
+    widthInt && heightInt
+      ? widthInt * heightInt < MIN_IMG_SIZE_FOR_PLACEHOLDER
+      : false
+  const shouldShowBlurryPlaceholder =
+    configPlaceholder === PlaceholderValue.BLURRY &&
+    placeholder === PlaceholderValue.BLURRY &&
+    priority &&
+    (layout === 'fill' || !tooSmallForBlurryPlaceholder)
```

Comment:
I think tooSmallForBlurryPlaceholder will already be false for layout fill images, because they don't have height and width